src/plugin.js

```diff
@@ -291,31 +249,39 @@ function panNumericalScale(scale, delta, panOptions) {
 	}
 
 	if (newMin >= rangeMin && newMax <= rangeMax) {
-		tickOpts.min = newMin;
-		tickOpts.max = newMax;
+		scaleOpts.min = newMin;
+		scaleOpts.max = newMax;
 	} else if (newMin < rangeMin) {
 		diff = prevStart - rangeMin;
-		tickOpts.min = rangeMin;
-		tickOpts.max = prevEnd - diff;
+		scaleOpts.min = rangeMin;
+		scaleOpts.max = prevEnd - diff;
 	} else if (newMax > rangeMax) {
 		diff = rangeMax - prevEnd;
-		tickOpts.max = rangeMax;
-		tickOpts.min = prevStart + diff;
+		scaleOpts.max = rangeMax;
+		scaleOpts.min = prevStart + diff;
 	}
 }
 
-function panTimeScale(scale, delta, panOptions) {
+function panCategoryScale(scale, delta, panOptions) {
```

Comment:
Can you move this back above `panNumericalScale`? It's difficult to tell what's change since its location in the file has changed

Reply:
Not possible, due to:

```
  115:1   warning  Function 'zoomCategoryScale' has a complexity of 15  complexity
  240:2   error    'panNumericalScale' was used before it was defined   no-use-before-define
  243:1   warning  Function 'panNumericalScale' has a complexity of 13  complexity
  413:41  warning  Function has a complexity of 11                      complexity
```

Reply:
If you want to use the `panNumericScale` into `panCategoryScale`, we must define it before, otherwise we must remove the rule of  `no-use-before-define`.
As you prefer.

Reply:
Ok. Let's not use `panNumericScale` in `panCategoryScale` to minimize the number of changes in this PR

Reply:
> 
> 
> Ok. Let's not use `panNumericScale` in `panCategoryScale` to minimize the number of changes in this PR

Done!